osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -325,6 +316,124 @@ private double computeFlashlightValue(ScoreInfo score, OsuDifficultyAttributes a
+
+            // If there's less then one miss detected - let combo-based miss count decide if this is FC or not
+            scoreBasedMissCount = Math.Max(scoreBasedMissCount, 1);
+
+            // Cap result by very harsh version of combo-based miss count
+            return Math.Min(scoreBasedMissCount, maximumMissCount);
+        }
+
+        /// <summary>
+        /// This function is a harsher version of current combo-based miss count, used to provide reasonable value for cases where score-based miss count can't do this.
+        /// </summary>
+        private double calculateMaximumComboBasedMissCount(OsuDifficultyAttributes attributes)
+        {
+            double missCount = 0;
+
+            // Consider that full combo is maximum combo minus dropped slider tails since they don't contribute to combo but also don't break it
+            // In classic scores we can't know the amount of dropped sliders so we estimate to 10% of all sliders on the map
+            double fullComboThreshold = attributes.MaxCombo - 0.1 * attributes.SliderCount;
+
+            if (scoreMaxCombo < fullComboThreshold)
+                missCount = Math.Pow(fullComboThreshold / Math.Max(1.0, scoreMaxCombo), 2.5);
+
+            // In classic scores there can't be more misses than a sum of all non-perfect judgements
+            missCount = Math.Min(missCount, totalImperfectHits);
+
+            return missCount;
+        }
+
+        private double calculateScoreAtCombo(OsuDifficultyAttributes attributes, double combo, double relevantComboPerObject, double scoreV1Multiplier)
+        {
+            double n = combo / relevantComboPerObject - 1;
+            double a = relevantComboPerObject - 1;
+
+            // The combo portion of ScoreV1 follows arithmetic progression
+            // Therefore, we calculate the combo portion of score using the combo per object and our current combo.
+            double comboScore = relevantComboPerObject > 0 ? (2 * a + (n - 1) * relevantComboPerObject) * n / 2 : 0;
+
+            // We then apply the accuracy and ScoreV1 multipliers to the resulting score.
+            comboScore *= accuracy * 300 / 25 * scoreV1Multiplier;
+
+            double objectsHit = (totalHits - countMiss) * combo / attributes.MaxCombo;
+
+            // Score also has a non-combo portion we need to create the final score value.
+            double nonComboScore = (300 + attributes.SliderNestedScorePerObject) * accuracy * objectsHit;
+
+            return comboScore + nonComboScore;
+        }
+
+        private double calculateScoreRelevantComboPerObject(OsuDifficultyAttributes attributes)
+        {
+            double comboScore = attributes.MaximumLegacyComboScore;
+
+            // We then reverse apply the ScoreV1 multipliers to get the raw value.
+            comboScore /= 300.0 / 25.0 * attributes.LegacyScoreBaseMultiplier;
+
+            double a = attributes.MaxCombo;
+            double b = comboScore;
```

Comment:
Can we just not use notation like that?

```
double comboPerObject = (attributes.MaxCombo - 2) * attributes.MaxCombo;
comboPerObject /= Math.Max(attributes.MaxCombo + 2 * (comboScore - 1), 1);
```
reads just fine

Reply:
The point of letters a and b to demonstrate that those are from arithmetic progression formula members, and not just random multiplications. I can change it of course.

Reply:
I think the comment below explains it enough without the need of using specific variable names